src/components/TextEditor/components/TextEditor.js

```diff
@@ -1,164 +1,141 @@
-  };
+  // const checkDirty = async () => {
+  //   if (!dirty) {
+  //     return;
+  //   }
+
+  //   try {
+  //     const programToUpdate = {};
+  //     programToUpdate[mostRecentProgram] = {
+  //       code: code,
+  //     };
 
-  updateCode = (editor, data, newCode) => {
+  //     await fetch.updatePrograms(uid, programToUpdate);
+  //     // TODO: add functionality to be able to tell whether the fetch failed
+  //   } catch (err) {
+  //     console.log(err);
+  //   }
+  // };
+
+  const updateCode = (editor, data, newCode) => {
     // if the code's not yet dirty, and the old code is different from the new code, make it dirty
-    if (!this.props.dirty && this.props.code !== newCode) {
-      this.props.dirtyCode(this.props.mostRecentProgram);
+    if (!dirty && code !== newCode) {
+      dirtyCode(mostRecentProgram);
     }
-    this.props.setProgramCode(this.props.mostRecentProgram, newCode);
+    setProgramCode(mostRecentProgram, newCode);
   };
 
-  setCurrentLine = (cm) => {
-    const { codeMirrorInstance, currentLine } = this.state;
+  const setCurrentLineManual = (cm) => {
     const { line } = cm.getCursor();
     if (codeMirrorInstance) {
       // removeLineClass removes the back highlight style from the last selected line
       codeMirrorInstance.removeLineClass(currentLine, 'wrap', 'selected-line');
       // addLineClass adds the style to the newly selected line
       codeMirrorInstance.addLineClass(line, 'wrap', 'selected-line');
     }
-    this.setState({ currentLine: line });
+    setCurrentLine(line);
   };
-
-  renderForkModal = () => (
-    <ReactModal
-      isOpen={this.state.showForkModal}
-      onRequestClose={this.closeForkModal}
-      className="fork-modal"
-      overlayClassName="profile-image-overlay"
-      ariaHideApp={false}
-    >
-      <h1 className="text-center">Fork This Sketch</h1>
-      {!(this.state.forking || this.state.forked) && (
-        <p className="text-center">Would you like to create your own copy of this sketch?</p>
-      )}
-      {this.state.forking ? (
-        <p className="text-center">Forking...</p>
-      ) : this.state.forked ? (
-        <div>
-          <p className="text-center">Sketch forked! Go to your sketches to see your new copy!</p>
-          <Button color="danger" size="lg" onClick={this.closeForkModal} block>
-            Close
-          </Button>
-          <Button color="success" size="lg" onClick={this.redirectSketch} block>
-            Go to Sketches
-          </Button>
-        </div>
-      ) : (
-        <div className="text-center">
-          <Button color="danger" size="lg" onClick={this.closeForkModal} block>
-            Cancel
-          </Button>
-          <Button color="success" size="lg" onClick={this.handleFork} block>
-            Fork
-          </Button>
-        </div>
-      )}
-    </ReactModal>
-  );
-
-  handleFork = async () => {
-    this.setState({ forking: true });
+  const redirectSketch = () => {
```

Comment:
nit (unnecessary, feel free to ignore): generally between new functions, we like to have a new line for code aesthetic! feel free to apply this throughout your file !
```suggestion

  const redirectSketch = () => {
```